pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -453,7 +499,9 @@ func (nim *nodeInfoManager) installDriverToCSINode(
 		}
 	}
 
-	if !specModified && !statusModified {
+	annotationModified := setMigrationAnnotation(nim.migratedPlugins, nodeInfo)
+
+	if !specModified && !statusModified && !annotationModified {
```

Comment:
can we remove status modified?

Reply:
Also if !specModified && annotationModified, will we continue and append the driver anyway? (duplicate entries)

Reply:
no duplicate entries, but the logic was a little bit wacky so i've updated it to be less confusing (imo) and the drivers will have more stable ordering.

Reply:
if `!specModified && annotationModified` the function will continue to run and append the new driver spec (which it had previously omitted in the for loop). We will not have duplicate entries.